libs/parallelism/futures/src/future_data.cpp

```diff
@@ -81,7 +83,64 @@ namespace hpx { namespace lcos { namespace detail {
     }
 
     ///////////////////////////////////////////////////////////////////////////
-    future_data_base<traits::detail::future_data_void>::~future_data_base() {}
+    future_data_base<traits::detail::future_data_void>::~future_data_base()
+    {
+        if (runs_child_ != threads::invalid_thread_id)
```

Comment:
What exactly does `runs_child_` represent?

Reply:
It refers to the thread that is supposed to deliver the value that will make the future ready. We could probably come up with a better name for this ;-) Any suggestions?